specs/consensus.md

```diff
@@ -0,0 +1,22 @@
+Consensus Rules
+===
+
+- [Consensus Rules](#consensus-rules)
+- [System Parameters](#system-parameters)
+- [Leader Selection](#leader-selection)
+- [Fork Choice](#fork-choice)
+- [Block Validity](#block-validity)
+
+# System Parameters
+
+| name                         | type     | value | description                                                      |
+| ---------------------------- | -------- | ----- | ---------------------------------------------------------------- |
+| `SHARE_SIZE`                 | `uint64` | `256` | Size of transaction and message shares, in bytes.                |
+| `SHARE_RESERVED_BYTES`       | `uint64` | `1`   | Bytes reserved at the beginning of each share.                   |
+| `INTERMEDIATE_ROOT_INTERVAL` | `uint64` | `1`   | Maximum number of transactions between intermediate state roots. |
+
+# Leader Selection
+
+# Fork Choice
+
+# Block Validity
```

Comment:
While it is straightforward to define _block validity_, we should prioritise specing out how validators are supposed to know _where_ to get the random chunks they require to verify the block is valid! This plays into specing the networking / p2p layer, too.

Reply:
Yep. This section should be relatively short, and only cover
1. proper formatting
1. availability (defer the work here to the networking spec)
1. potential state transitions to the validator set

Reply:
Hmm, maybe we should divide the spec into two parts: 1) the core consensus critical parts of the spec (mesage, tree formats etc), and 2) the non-consensus critical networking protocols.

Reply:
Yes, the above discussion had the separation of local consensus rules from networking in mind. The separate networking spec will be in https://github.com/LazyLedger/lazyledger-specs/blob/adlerjohn-initial_data_structures/specs/networking.md.